gpbft/gpbft.go

```diff
@@ -1026,35 +1027,47 @@ type convergeState struct {
 	// Chains indexed by head CID
 	values map[TipSet]ECChain
 	// Tickets provided by proposers of each chain.
-	tickets map[TipSet][]Ticket
+	tickets map[TipSet]map[ActorID]Ticket
 }
 
 func newConvergeState() *convergeState {
 	return &convergeState{
 		values:  map[TipSet]ECChain{},
-		tickets: map[TipSet][]Ticket{},
+		tickets: map[TipSet]map[ActorID]Ticket{},
 	}
 }
 
 // Receives a new CONVERGE value from a sender.
-func (c *convergeState) Receive(value ECChain, ticket Ticket) error {
+func (c *convergeState) Receive(sender ActorID, value ECChain, ticket Ticket) error {
 	if value.IsZero() {
 		return fmt.Errorf("bottom cannot be justified for CONVERGE")
 	}
 	key := value.Head()
 	c.values[key] = value
-	c.tickets[key] = append(c.tickets[key], ticket)
+
+	if _, ok := c.tickets[key]; !ok {
+		c.tickets[key] = make(map[ActorID]Ticket)
+	}
+
+	c.tickets[key][sender] = ticket
 
 	return nil
 }
 
-func (c *convergeState) findMinTicketProposal() ECChain {
-	var minTicket Ticket
+func (c *convergeState) findMinTicketProposal(table PowerTable) ECChain {
+	var minTicket *big.Float
```

Comment:
Let us just use the max function then instead of min? This way we just multiply. This must be changed in the FIP though. Using min or max has no impact anywhere else.

Reply:
As discussed, max seems fine. You can go back to min if implemented with truncating integer division, too.